src/server/identity/server/connector.go

```diff
@@ -10,25 +10,34 @@ import (
 	"github.com/dexidp/dex/pkg/log"
 )
 
+const localhostIdentityServer = "localhost:30658"
+
 var (
 	_ connector.CallbackConnector = &placeholder{}
 )
 
 // placeholder is a fake Dex connector which redirects the user to a static page with instructions.
 // This is necessary because Dex won't start the web server unless a connector is configured.
-type placeholderConfig struct{}
+type placeholderConfig struct {
+	LocalhostIssuer bool
+}
 
-func (placeholderConfig) Open(id string, logger log.Logger) (connector.Connector, error) {
-	return &placeholder{}, nil
+func (p placeholderConfig) Open(id string, logger log.Logger) (connector.Connector, error) {
+	return &placeholder{LocalhostIssuer: p.LocalhostIssuer}, nil
 }
 
-type placeholder struct{}
+type placeholder struct {
+	LocalhostIssuer bool
+}
 
-func (*placeholder) LoginURL(s connector.Scopes, callbackURL, state string) (string, error) {
+func (p *placeholder) LoginURL(s connector.Scopes, callbackURL, state string) (string, error) {
 	u, err := url.Parse(callbackURL)
 	if err != nil {
 		return "", fmt.Errorf("failed to parse callbackURL %q: %v", callbackURL, err)
 	}
+	if p.LocalhostIssuer {
+		u.Host = localhostIdentityServer
+	}
```

Comment:
I'm wondering if we could get away with a relative redirect to avoid this? Not a big deal either way.